### DISCUSSION TITLE
训练paddlev4的文本行检测时，在每个epoch会对每个数据集进行重新采样吗？

### DISCUSSION THREAD

**Discussion by GrayChan813** [PaddlePaddle/PaddleOCR#13612] [Q&A]
在config里面，我们将label_file_list和ratio_list设置为一个列表对象，以方便加载多个训练集或验证集，每一个ratio_list的里面的比例代表label_file_list对应的数据集的采样比例。

所以我很好奇，在每个epoch开始之前，会不会重新对label_file_list里面的数据集按照ratio_list重新采样一遍，以达到在训练足够多轮数的情况下使得所有的训练或验证样本都能够被使用到。还是说，按照ratio_list进行采样的操作只会进行一次，随后的所有epoch训练或验证都用的是一开始采样的数据？

在阅读代码的时候，我发现./tools/program.py的第293-302行有一段这样的代码：
    for epoch in range(start_epoch, epoch_num + 1):
        if train_dataloader.dataset.need_reset:
            train_dataloader = build_dataloader(
                config, "Train", device, logger, seed=epoch
            )
            max_iter = (
                len(train_dataloader) - 1
                if platform.system() == "Windows"
                else len(train_dataloader)
            )

这里train_dataloader.dataset.need_reset设置为True的话，是否能够实现在每个epoch对数据集重新进行采样。如果是的话，那么paddlev4是不是就默认只会在一开始进行一次的采样呢？

感谢各位大佬！！！希望有人能解答我的疑惑！！！谢谢！！！

**Comment by UserWangZz** [comment]
https://github.com/PaddlePaddle/PaddleOCR/blob/6dc021115cb16bd6673a4e28ab267f0465c14c4f/ppocr/data/simple_dataset.py#L26-L58
根据上面数据集初始化代码第58行，会将ratio_list中所有小于1的数据集的need_reset置为True，结合你给出的代码，我理解应该会对数据集进行重新采样。

**Reply by GrayChan813** [reply]
感谢！看了展示的代码那就是每个epoch之前都会重新采样了！十分感谢您解答我的疑惑！
